drivers/hmis/app/models/hmis/hud/processors/geolocation_processor.rb

```diff
@@ -0,0 +1,46 @@
+#  Copyright 2016 - 2024 Green River Data Analysis, LLC
+#
+#  License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+#
+
+module Hmis::Hud::Processors
+  class GeolocationProcessor < Base
+    def factory_name
+      :clh_location_factory
+    end
+
+    def schema
+      nil
+    end
+
+    def process(field, value)
+      attribute_name = ar_attribute_name(field)
+
+      if attribute_name == 'coordinates'
+        attribute_value = JSON.parse(value, symbolize_names: true)
+        latitude = attribute_value[:latitude]
+        longitude = attribute_value[:longitude]
+        not_collected_reason = attribute_value[:notCollectedReason]
+        raise ArgumentError, 'Geolocation coordinates in unexpected format' unless (latitude && longitude) || not_collected_reason
+
+        return @processor.send(factory_name).destroy if not_collected_reason
+
+        @processor.send(factory_name).assign_attributes(lat: latitude, lon: longitude)
+      else
+        @processor.send(factory_name).assign_attributes(attribute_name => value)
```

Comment:
Do we ever hit this code? If we don't expect any other fields, maybe we should raise an error.